vmdb/app/controllers/ems_infra_controller.rb

```diff
@@ -33,41 +33,56 @@ def scaling
       return
     end
 
-    @count_parameters = @stack.parameters.select { |x| x.name.include?('::count') }
+    @orig_parameters = @stack.parameters.select { |x| x.name.include?('::count') }
 
     return unless params[:scale]
 
-    scale_parameters = params.select { |k, _v| k.include?('::count') }
-    assigned_hosts = scale_parameters.values.sum(&:to_i)
+    form_parameters = params.select { |k, _v| k.include?('::count') }
+
+    validate_selected_hosts_not_exceed_available_hosts(form_parameters)
+
+    changed_parameters = []
+    return_message = _("Scaling")
+
+    select_changed_parameters_and_return_message(form_parameters, changed_parameters, return_message)
+    validate_a_change_was_made(changed_parameters)
+
+    return unless changed_parameters.length > 0
```

Comment:
better if the check is done inside validate_a_change_was_made(changed_parameters)

so you can write 

return unless validate_a_change_was_made(changed_parameters)